### PR TITLE
feat(vm): EVM active precompiles

### PR DESCRIPTION
**Pull request by fedekunze**
This PR implements custom active precompiles from the EVM, which can be set by calling the `WithPrecompiles` function

### CODE REVIEW

File: core/vm/contracts.go [hunk 1]
```diff
@@ -141,6 +143,87 @@ func ActivePrecompiles(rules params.Rules) []common.Address {
...
+// ActivePrecompiles returns the precompiles enabled with the current configuration.
+//
+// NOTE: The rules argument is ignored as the active precompiles can be set via the WithPrecompiles
+// method according to the chain rules from the current block context.
+func (evm *EVM) ActivePrecompiles(_ params.Rules) []common.Address {
+	return evm.activePrecompiles
+}
+
+// Precompile returns returns a precompiled contract for the given address. This
+// function returns false if the address is not a registered precompile.
```
Comment (Vvaradinov) [inline core/vm/contracts.go:171]:
```suggestion
// Precompile returns a precompiled contract for the given address. This
// function returns false if the address is not a registered precompile.
```

File: core/vm/evm.go [hunk 2]
```diff
@@ -120,6 +104,10 @@ type EVM struct {
 	// available gas is calculated in gasCall* according to the 63/64 rule and later
 	// applied in opCall*.
 	callGasTemp uint64
+	// precompiles defines the precompiles contracts used by the EVM
```
Comment (Vvaradinov) [inline core/vm/evm.go:107]:
```suggestion
	// precompiles defines the precompiled contracts used by the EVM
```

**Review by Vvaradinov — APPROVED** [review]
LGTM!

**Comment by fedekunze** [comment]
@mergifyio backport release/1.10

**Pull request merged**